packages/contracts/contracts/proxies/ProxyFactory.sol

```diff
@@ -18,11 +17,57 @@ contract ProxyFactory {
     returns (Proxy proxy)
   {
     proxy = new Proxy(masterCopy);
-    if (data.length > 0) {
+    if (data.length > 0)
+      // solium-disable-next-line security/no-inline-assembly
       assembly {
         if eq(call(gas, proxy, 0, add(data, 0x20), mload(data), 0, 0), 0) { revert(0, 0) }
       }
+    emit ProxyCreation(proxy);
+  }
+
+  /// @dev Allows to retrieve the runtime code of a deployed Proxy. This can be used to check that the expected Proxy was deployed.
+  function proxyRuntimeCode() public pure returns (bytes memory) {
+    return type(Proxy).runtimeCode;
+  }
+
+  /// @dev Allows to retrieve the creation code used for the Proxy deployment. With this it is easily possible to calculate predicted address.
+  function proxyCreationCode() public pure returns (bytes memory) {
+    return type(Proxy).creationCode;
+  }
+
+  /// @dev Allows to create new proxy contact and execute a message call to the new proxy within one transaction.
```

Comment:
Should be more specific about this function wrt a contract creation vs. address determination.